pkgs/by-name/si/simutrans/package.nix

```diff
@@ -2,196 +2,439 @@
   lib,
   stdenv,
   fetchurl,
+  subversion,
+  fetchsvn,
+  fetchzip,
   pkg-config,
-  unzip,
   zlib,
+  zstd,
   libpng,
   bzip2,
-  SDL,
-  SDL_mixer,
-  buildEnv,
-  config,
-  runtimeShell,
+  SDL2,
+  SDL2_mixer,
+  fontconfig,
+  miniupnpc,
+  symlinkJoin,
+  makeWrapper,
+  runCommand,
+  cabextract,
 }:
-
 let
-  # Choose your "paksets" of objects, images, text, music, etc.
-  paksets = config.simutrans.paksets or "pak64 pak64.japan pak128 pak128.britain pak128.german";
-
-  result = withPaks (
-    if paksets == "*" then
-      lib.attrValues pakSpec # taking all
-    else
-      map (name: pakSpec.${name}) (lib.splitString " " paksets)
-  );
-
-  ver1 = "121";
-  ver2 = "0";
-  ver3 = "";
-  version = "${ver1}.${ver2}${lib.optionalString (ver3 != "") ".${ver3}"}";
-  ver_dash = "${ver1}-${ver2}${lib.optionalString (ver3 != "") "-${ver3}"}";
-
-  binary_src = fetchurl {
-    url = "mirror://sourceforge/simutrans/simutrans/${ver_dash}/simutrans-src-${ver_dash}.zip";
-    sha256 = "1f463r6kr5ig0zd3mncc74k93xbjywsq3d06j5r17831jyc9bzb9";
+  fetchcab =
+    args:
+    let
+      src = fetchurl args;
+    in
+    runCommand "source-cab" { nativeBuildInputs = [ cabextract ]; } ''
+      cabextract ${src} -d $out
+    '';
```

Comment:
Could we use postFetch inside of fetchurl to extract the files?

Reply:
probably! never knew that existed. i'll see what i can do

Reply:
Keeping them packed makes FOD also easier. Whatever suites you.